mne/io/edf/edf.py

```diff
@@ -1112,23 +1116,33 @@ def _read_annot(annot, annotmap, sfreq, data_length):
     stim_channel : ndarray
         An array containing stimulus trigger events.
     """
-    pat = '([+/-]\\d+.\\d+),(\\w+)'
-    annot = open(annot).read()
-    triggers = re.findall(pat, annot)
-    times, values = zip(*triggers)
+    from io import open  # python 2 backward compatible open
```

Comment:
if you use `import codecs` then `with codecs.open(...)` you don't need to shadow the builtin or nest the import

alternatively at the top of the file you could also do `from io import open as io_open` or so and then use `with io_open` here.